### ISSUE TITLE
If Window Image Cache Lifespan is 0, Don't Cache

### ISSUE THREAD

**Issue by ShlomoCode** [ejbills/DockDoor#241]
If Window Image Cache Lifespan is set to 0 (ie a completely fresh image every time), there is no need to save image cache, thus saving the user RAM.
@ejbills

**Comment by ShlomoCode** [comment]
Dependent on #238

**Comment by stevehoek** [comment]
I have my cache setting to the lowest non-zero setting and it still shows old previews.  How can you force it to refresh dock preview without restarting DockDoor?  I've activated the window, minimized and restored, etc. and it still shows an old preview image

**Comment by ShlomoCode** [comment]
@stevehoek This sounds like an unrelated bug, please open a new bug issue and fill in all the required information

**Comment by ejbills** [comment]
> I have my cache setting to the lowest non-zero setting and it still shows old previews.  How can you force it to refresh dock preview without restarting DockDoor?  I've activated the window, minimized and restored, etc. and it still shows an old preview image
> 
> 
> 
> 

Latest update should fix.

**Comment by ShlomoCode** [comment]
@ejbills Why did it close?

**Comment by ejbills** [comment]
> @ejbills Why did it close?

We need to cache, as that is how the app is tracking windows across desktop spaces and providing minimized/hidden previews. With the current system, we can't not cache—or if we did, it'd mean losing the aforementioned features. 

Did you have ideas to get around this? I am open to suggestions.

**Comment by stevehoek** [comment]
I think the idea was if it was set to not cache, it would generate a fresh preview every time so it would never be stale.  That may hit performance too much, idk.

**Comment by ejbills** [comment]
@stevehoek, yes, setting the cache to 0 seconds essentially achieves this. However, it’s crucial to consider the following:
- Screenshotting windows in other desktop spaces requires extensive workarounds and is a hassle to implement.
- Minimized and hidden application windows are also _impossible_ to screenshot (with official screen capture methods), with the same limitation.

We “cache” windows to display their images when they fall into these three categories. In such cases, we use the cached image for the window’s representation. Based on the user-set cache lifespan, all window previews are updated as soon as possible. So, with the cache set to 0, we could implement not caching—but with the caveat of having to redesign a significant portion of the backend code, which may not be the most efficient use of time, considering this only results in a slight reduction in RAM usage. 

Hoping to hear more opinions on this.

**Comment by ShlomoCode** [comment]
Yes now I remember, this was written when there was a plan to use one undocumented API for live images of minimized windows, another space, etc. If we plan to stick with only official API, disabling caching could hurt the user experience
